dali/operators/util/normal_distribution_op.cc

```diff
@@ -0,0 +1,103 @@
+                .NumInput(0, 1)
+                .NumOutput(detail::kNumOutputs)
+                .AddOptionalArg(detail::kMean, R"code(Mean value of the distribution)code",
+                                0.f, true)
+                .AddOptionalArg(detail::kStddev,
+                                R"code(Standard deviation of the distribution)code",
+                                1.f, true)
+                .AddOptionalArg(detail::kShape,
+                                R"code(Shape of single output tensor in a batch)code",
+                                detail::kShapeDefaultValue, true)
+                .AddOptionalArg(arg_names::kDtype, R"code(Data type for the output)code",
+                                DALI_FLOAT);
+
+DALI_REGISTER_OPERATOR(NormalDistribution, NormalDistributionCpu, CPU);
+
+
+#define NORM_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float, double)  // NOLINT
+
+
+bool NormalDistributionCpu::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                      const workspace_t<CPUBackend> &ws) {
+  AcquireArguments(ws);
+  output_desc.resize(detail::kNumOutputs);
+  output_desc[0].shape = GetOutputShape(ws);
+  TYPE_SWITCH(dtype_, type2id, DType, NORM_TYPES, (
+          {
+            TypeInfo type;
+            type.SetType<DType>(dtype_);
+            output_desc[0].type = type;
+          }
+  ), DALI_FAIL(make_string("Unsupported output type: ", dtype_)))  // NOLINT
+  return true;
+}
+
+
+void NormalDistributionCpu::AssignSingleValueToOutput(workspace_t<CPUBackend> &ws) {
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  TYPE_SWITCH(dtype_, type2id, DType, NORM_TYPES, (
+          distribution_t distribution(mean_[0], stddev_[0]);
+          for (int sample_id = 0; sample_id < batch_size_; ++sample_id) {
+              auto ptr = output[0].mutable_data<DType>();
+              *ptr = ConvertSat<DType>(distribution(rng_));
+          }
+  ), DALI_FAIL(make_string("Unsupported output type: ", dtype_)))  // NOLINT
+}
+
+
+void NormalDistributionCpu::AssignTensorToOutput(workspace_t<CPUBackend> &ws) {
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  auto &tp = ws.GetThreadPool();
+  TYPE_SWITCH(dtype_, type2id, DType, NORM_TYPES, (
+            for (int sample_id = 0; sample_id < batch_size_; ++sample_id) {
+                distribution_t distribution(mean_[sample_id], stddev_[sample_id]);
+                auto ptr = output[sample_id].mutable_data<DType>();
+                for (long j = 0; j < volume(output[sample_id].shape()); j++) {  // NOLINT (long)
+                    ptr[j] = ConvertSat<DType>(distribution(rng_));
+                }
+            }
+  ), DALI_FAIL(make_string("Unsupported output type: ", dtype_)))  // NOLINT
+}
+
+
+void NormalDistributionCpu::RunImpl(workspace_t<CPUBackend> &ws) {
+  if (this->single_value_in_output_) {
+    AssignSingleValueToOutput(ws);
+  } else {
+    AssignTensorToOutput(ws);
+  }
+}
+
+
+#undef NORM_TYPES
```

Comment:
no need to do this, I believe

Reply:
I'd do this, since I wouldn't like it to be used anywhere else in this compilation unit. It's just in case ;)